pilot/pkg/serviceregistry/federation/controller.go

```diff
@@ -504,21 +504,15 @@ func (c *Controller) getEgressServiceAddrs() ([]model.NetworkGateway, []string)
 	var addrs []model.NetworkGateway
 	var sas []string
 	for _, subset := range endpoints.Subsets {
-		for index, address := range subset.Addresses {
-			if subset.Ports[index].Port == common.DefaultFederationPort {
```

Comment:
As the documentation of [Endpoint API](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/endpoints-v1/) says, a subset of endpoints contains:
```
IP addresses which offer the related ports that are marked as ready.
```
so there is no need to look for a port at index of an address.

Reply:
Correct. Ports are a property of the subset, not of the address.